src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,1052 @@
+        # dec_features, decoder_cached_states, dec_hidden, dec_attn
+        decoder_outputs = self.decoder.forward(
+            decoder_input_ids,
+            encoder_outputs[0],
+            attention_mask,
+            decoder_attention_mask,
+            decoder_cached_states=decoder_cached_states,
+        )
+        # Attention and hidden_states will be [] or None if they aren't needed
+        decoder_outputs = _filter_out_falsey_values(decoder_outputs)  # type: tuple
+        assert isinstance(decoder_outputs[0], torch.Tensor)
+        encoder_outputs = _filter_out_falsey_values(encoder_outputs)  # type: tuple
+        return decoder_outputs + encoder_outputs
+
+
+@add_start_docstrings(
+    "The bare BART Model with a language modeling head", BART_START_DOCSTRING, BART_INPUTS_DOCSTRING,
+)
+class BartForMaskedLM(PretrainedBartModel):
+    r"""
+        **lm_labels**: (`optional`) ``torch.LongTensor`` of shape ``(batch_size, sequence_length)``:
+            Labels for computing the masked language modeling loss.
+            Indices should either be in ``[0, ..., config.vocab_size]`` or -100 (see ``input_ids`` docstring).
+            Tokens with indices set to ``-100`` are ignored (masked), the loss is only computed for the tokens with labels
+            in ``[0, ..., config.vocab_size]``.
+
+    Outputs: `Tuple` comprising various elements depending on the configuration (config) and inputs:
+        **loss**: (`optional`, returned when ``lm_labels`` is provided) ``torch.FloatTensor`` of shape ``(1,)``:
+            Masked language modeling loss.
+        **prediction_scores**: ``torch.FloatTensor`` of shape ``(batch_size, sequence_length, config.vocab_size)``
+            Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+        **hidden_states**: (`optional`, returned when ``config.output_hidden_states=True``)
+            list of ``torch.FloatTensor`` (one for the output of each layer + the output of the embeddings)
+            of shape ``(batch_size, sequence_length, hidden_size)``:
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        **attentions**: (`optional`, returned when ``config.output_attentions=True``)
+            list of ``torch.FloatTensor`` (one for each layer) of shape ``(batch_size, num_heads, sequence_length, sequence_length)``:
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention heads.
+
+    Examples::
+
+        tokenizer = BartTokenizer.from_pretrained('bart-large')
+        model = BartForMaskedLM.from_pretrained('bart-large')
+        input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)  # Batch size 1
+        outputs = model(input_ids=input_ids, lm_labels=input_ids)
+        loss, prediction_scores = outputs[:2]
+
+    """
+    base_model_prefix = "model"
```

Comment:
Additional inputs/outputs not detailed in the START/INPUTS docstrings are now added to the forward method as well, cf. [BertForPreTraining](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bert.py#L830-L877).

Reply:
Also, mind how the format changed from "**attentions**: (`optional`, returned when ``config.output_attentions=True``)" to "attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``config.output_attentions=True``):".

I believe you can copy and paste most of it.